### ISSUE TITLE
Force TP loophole.

### ISSUE THREAD

**Issue by Davros-and-Babble** [OpenCollarTeam/OpenCollar#112]
What version of OpenCollar are you using? 7.0 dev 4

As we know, griefers will grief, and collars are no exception to that.  I've heard about issues like these back when I ran an sim where open RLV was encouraged, and griefers would find ways to abuse open access collars by teleporting wearers to different sim (among others things - _such as commands through relays, or grabbling alot of leashes and racing across the sim_).  I saw these TP loopholes first hand when I was testing today.

The issues:

- For Trusted (without refuse TP checked), or group members in group mode, or everyone in open mode: Force TP  is not allowed.  However:
  - You can still force TP the wearer through the Terminal (tpto:sim/x/y/z=force)
  - You can still force TP the wearer by using bookmark apps (Save>put in SLURL>put in name, then click the new URL).  … Even when the Relay is off

- For Trusted (without refuse TP checked) … but NOT group mode or open mode:
  - The OpenCollar Remote HUD has a bookmarks menu allows trusted to force TP them, regardless of the refuse TP exception being unchecked.

- The bookmarks in the OpenCollar Apps, and the bookmarks in the Remote HUD appears to be two separate apps, that can have different lists of bookmarks. This is probably as expected, but adding it here just in case.
	

TLDR;
I tested Group/Open/Capture modes today mostly to figure out what could and could not be done in all cases (regardless of the TP loopholes), and wanted to share my findings here in case there's any interested parties:

- Capture mode appears to have full access to everything, including settings.  The only thing you cannot do that is add or remove owners and trusted, including the "OwnSelf" button.  This appears to be expected behavior.

- Group Mode and Open mode appear to have the same access.  
  - They can **NOT**: Lock, Leash Configuration, Turn RLV Off/On, Any Access Menu changes, Any Rely Menu changes, or Force TP.   
  - They also have **limited** access to Apps.

**Comment by CorwinDavidson** [comment]
@Davros-and-Babble  
Could you explain what you mean by "Terminal"

The oc_bookmarks script had problems where it was not correctly refusing GROUP/ Public Access TP commands.  It would report No Access yet still perform the command.  This is now fixed on Github.

**Comment by Davros-and-Babble** [comment]
@CorwinDavidson Terminal is the lower left hand button on the RLV menu
     /1 <prefex> Menu   ->  RLV>Terminal
     https://gyazo.com/273eff52bdedb39bffd6771c24cf9ac9
And oddly I didnt have issues with the hud force tping for group/open.  It was just with Trusted being able to force TP:

- For Trusted (without refuse TP checked) … but NOT group mode or open mode:

  - The OpenCollar Remote HUD has a bookmarks menu allows trusted to force TP them, regardless of the refuse TP exception being unchecked.

**Comment by IsabeauConnaught** [comment]
@Davros-and-Babble terminal is part of latest dev version, so if @CorwinDavidson runs stable released 7.0, or dev2/dev3 he may not know about this feature

**Comment by Davros-and-Babble** [comment]
Makes sense!  So on the stable released 7.0, you can access the terminal via the command even thought he button doesnt exist.  **prefix Terminal**

**Comment by IsabeauConnaught** [comment]
ohh.. i didnt know terminal is available via chat command. well, one learns something new every day

**Comment by SatomiAhn** [comment]
@Davros-and-Babble: AFAIK, all of this is intended behavior.
Indeed, I believe you misinterpreted "refuse TP" (if you mean the button in the Exceptions submenu). This option only concerns TP invitations (the ones you send through the viewer UI). It is not meant to block RLV commands.

That said, we cannot expect the user to guess all underlying mechanisms. Maybe this button should be made more understandable (starting its very presence inside the "Exceptions" submenu. To what is this supposed to be an exception?).

(What follows actually deserves its own feature request:) Also, to make things consistent, the collar needs an option to restrict access to the terminal (the relay already has such options). One possible implementation: a white list of allowed RLV commands for each auth level. A collar menu is not needed for this: let's just have default white lists in the settings notecard, then advanced users can tweak with them if they wish to.

**Comment by Davros-and-Babble** [comment]
@SatomiAhn  You're right, this pretty much is the way RLV works.  It just seems odd that if the option is given to stop forced tps from trusted users by unchecking "refuseTP" which I believe must use the @acceptTP rem command, that the user can still use other RLV commands to force TP.  And perhaps this is better submitted to the team working on RLV :)  But on the other side, it seems RLV was just meant as a general set of commands which do not govern themselves, but expect the device to govern the rules of what commands are issues and what commands are not issued.  If we are looking from the latter perspective, doesn't it make sense that if an option is given to refuseTP, that it will work that way in all cases since the collar is governing what RLV commands should be given?  I suppose we could just say its restricting remote Teleportation and leave it at that, since I believe all the other options require the person who is teleporting to be within 20m.

**Comment by GarvinTwine** [comment]
For your info:
The Terminal feature was introduced to OpenCollar version 4.0 by VirtualDisgrace when we added the rlvsuite script from our distro to the general opencollar distro.

**Comment by zontreck** [comment]
Is this still a problem?

**Comment by zontreck** [comment]
Terminal now has access restrictions, or at least.. the ability to do so:) and the bookmarks app is only accessible to owner, wearer and trusted.

**Comment by zontreck** [comment]
As for a HUD.. I'd have to check but i do not think that it would allow group, nor public to do a loophole using that method.
